services/api/src/typeDefs.js

```diff
@@ -1083,6 +1089,10 @@ const typeDefs = gql`
     """
     projectByName(name: String!): Project
     """
+    Returns all Environment Objects for a specified Kubernetes
+    """
+    environmentsByKubernetes(kubernetes: KubernetesInput!, order: EnvOrderType, createdAfter: String): [Environment]
```

Comment:
The `allEnvironments` query has a filter for the `type` too. I think it could be useful to add this filter to this resolver too. Then you could do all production environments on cluster X